### PR TITLE
lFeat/leaderboard datagrid cr fixes

### PR DESCRIPTION
**Pull request by adrian-oleskiewicz**
… and pinned column by custom hook / sorting hack replaced with mui api / @mui/x-data-grid reinstalled properly

## Related issues
[Keywords for linking issues](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests)

### CODE REVIEW

File: packages/apps/dashboard/ui-2024/src/features/Leaderboard/components/DataGridWrapper.tsx [hunk 1]
```diff
@@ -4,145 +4,134 @@ import { handleErrorMessage } from '@services/handle-error-message';
...
+        loading={status === 'pending'}
+        rows={visibleRows}
+        columns={customizedColumns}
+        autosizeOptions={{
+          expand: true,
+        }}
+        onSortModelChange={handleSortModelChange}
+        initialState={{
+          sorting: {
+            sortModel: [{ field: 'amountStaked', sort: 'desc' }],
```
Comment (adrian-oleskiewicz) [inline packages/apps/dashboard/ui-2024/src/features/Leaderboard/components/DataGridWrapper.tsx:56]: default sorting changed

File: packages/apps/dashboard/ui-2024/src/features/Leaderboard/components/DataGridWrapper.tsx [hunk 2]
```diff
@@ -4,145 +4,134 @@ import { handleErrorMessage } from '@services/handle-error-message';
...
+        hideFooter
+        disableVirtualization={isMobile}
+        apiRef={apiRef}
+        loading={status === 'pending'}
+        rows={visibleRows}
+        columns={customizedColumns}
+        autosizeOptions={{
+          expand: true,
+        }}
+        onSortModelChange={handleSortModelChange}
```
Comment (adrian-oleskiewicz) [inline packages/apps/dashboard/ui-2024/src/features/Leaderboard/components/DataGridWrapper.tsx:53]: handler fn from custom hook

File: packages/apps/dashboard/ui-2024/src/features/Leaderboard/components/DataGridWrapper.tsx [hunk 3]
```diff
@@ -4,145 +4,134 @@ import { handleErrorMessage } from '@services/handle-error-message';
...
+          '& .MuiDataGrid-columnHeader:focus': {
+            outline: 'none',
+          },
+          '& .MuiDataGrid-virtualScroller': {
+            position: 'relative',
+          },
+          '& .MuiDataGrid-filler': {
+            display: 'none',
+          },
+          ...pinnedColSx,
```
Comment (adrian-oleskiewicz) [inline packages/apps/dashboard/ui-2024/src/features/Leaderboard/components/DataGridWrapper.tsx:131]: styles for pinnedCol from custom hooks

File: packages/apps/dashboard/ui-2024/src/features/Leaderboard/hooks/useDataGrid.tsx [hunk 4]
```diff
@@ -16,196 +16,175 @@ import { ReputationLabel } from '../components/ReputationLabel';
 import { TextCell } from '../components/TextCell';
```
Comment (adrian-oleskiewicz) [inline packages/apps/dashboard/ui-2024/src/features/Leaderboard/hooks/useDataGrid.tsx:16]: pinned col and row order num logic moved to custom hook useDatagridOtions

File: packages/apps/dashboard/ui-2024/.prettierrc [hunk 5]
```diff
@@ -3,6 +3,6 @@
 	"semi": true,
```
Comment (adrian-oleskiewicz) [inline packages/apps/dashboard/ui-2024/.prettierrc:3]: indent changed to spaces, we need to apply it to other files in this package. I didn't do it to show only related changes to CR fixes

**Review by mgacek-blockydevs — APPROVED** [review]

**Pull request merged**